src/main/java/com/bravo/user/dao/model/mapper/ResourceMapper.java

```diff
@@ -22,15 +23,21 @@ public ResourceMapper(MapperFacade mapperFacade) {
   }
 
   public UserReadDto convertUser(final User user){
-    final UserReadDto dto = mapperFacade.map(user, UserReadDto.class);
-
-    String name;
-    if(user.getMiddleName() != null && !user.getMiddleName().trim().isEmpty()){
-      name = String.format("%s %s %s", user.getFirstName(), user.getMiddleName(), user.getLastName());
-    } else {
-      name = String.format("%s %s", user.getFirstName(), user.getLastName());
-    }
-    dto.setName(name);
-    return dto;
+      final UserReadDto dto = mapperFacade.map(user, UserReadDto.class);
+
+      String name;
+      if (user.getMiddleName() != null && !user.getMiddleName().trim().isEmpty()) {
+          name = String.format("%s %s %s", user.getFirstName(), user.getMiddleName(), user.getLastName());
+      } else {
+          name = String.format("%s %s", user.getFirstName(), user.getLastName());
+      }
+      dto.setName(name);
+
+      //add auth email and role to user dto
+      if (user.getAuth() != null) {
```

Comment:
I agree with this approach.  This enables a user to not have to have an entry in the authentication table.  Is there a way in code to enforce every user to also have a login?

Reply:
Honestly, with the db constraint I added that ties user to auth, I don't think this null check is needed, but I added it cause I wasn't sure if a null pointer exception would be thrown from any of the tests that build out the user entities.  Now that I think about it, I don't think an exception would have been thrown and I also could have added the NotNull annotation to role and email to enforce them at the user bean level and optional = false to the auth mapping in the user entity.